server/server.go

```diff
@@ -328,11 +328,12 @@ func (s *Server) getListenerFromURL(u *url.URL) (func() error, func() error, err
 		return nil, nil, fmt.Errorf("invalid url scheme %q", u.Scheme)
 	}
 
+	socketPath := u.Host + u.Path
+
 	// Remove domain socket file in case it already exists.
-	os.Remove(u.Path)
+	os.Remove(socketPath)
```

Comment:
[nit] ignoring the potentially returned error is ok here, I suppose? (I'm lacking context, just knee-jerking errcheck review comments. 😄)